base/iobuffer.jl

```diff
@@ -29,14 +104,20 @@ function GenericIOBuffer(data::T, readable::Bool, writable::Bool, seekable::Bool
                          maxsize::Integer) where T<:AbstractVector{UInt8}
     GenericIOBuffer{T}(data, readable, writable, seekable, append, maxsize)
 end
+
+# For this method, we use the underlying Memory of the vector. Therefore, we need to set the,
+# ptr and size accordingly, so the buffer only uses the part of the memory that the vector does.
 function GenericIOBuffer(data::Vector{UInt8}, readable::Bool, writable::Bool, seekable::Bool, append::Bool,
                          maxsize::Integer)
     ref = data.ref
-    buf = GenericIOBuffer(ref.mem, readable, writable, seekable, append, maxsize)
+    mem = ref.mem
+    len = length(data)
```

Comment:
If `len < maxsize < length(mem)` could this trip the `"maxsize must not be smaller than data length"` error in an unexpected way?

Reply:
Good catch. I've addressed this by setting `maxsize` to `length(mem)` in that case. That's not ideal - since if the user `take!`s the buffer, and then re-initializes it, it may grow to larger than expected.
However, I can't think of a better way to handle it. Alternatively, we could allow `maxsize` to be smaller than the buffer length in that particular case.